### PR TITLE
SC-559 MakeString function

### PR DESCRIPTION
**Pull request by xrtm000**

### CODE REVIEW

File: lang/shared/src/main/scala/com/wavesplatform/lang/v1/evaluator/ctx/impl/PureContext.scala [hunk 1]
```diff
@@ -514,6 +514,21 @@ object PureContext {
       )
   }
 
+  lazy val makeString: BaseFunction[NoContext] =
+    NativeFunction("makeString", 30, MAKESTRING, listString, ("list", LIST(STRING)), ("separator", STRING)) {
+      case ARR(list) :: CONST_STRING(separator) :: Nil =>
+        val expectedStringSize =
```
Comment (potan) [inline lang/shared/src/main/scala/com/wavesplatform/lang/v1/evaluator/ctx/impl/PureContext.scala:520]: Could we check limits by array weight without calculation utf8 bytes?

Comment (xrtm000) [reply lang/shared/src/main/scala/com/wavesplatform/lang/v1/evaluator/ctx/impl/PureContext.scala]: Fixed

File: lang/shared/src/main/scala/com/wavesplatform/lang/v1/evaluator/ctx/impl/PureContext.scala [hunk 2]
```diff
@@ -514,6 +514,18 @@ object PureContext {
       )
   }
 
+  lazy val makeString: BaseFunction[NoContext] =
+    NativeFunction("makeString", 30, MAKESTRING, listString, ("list", LIST(STRING)), ("separator", STRING)) {
+      case ARR(list) :: CONST_STRING(separator) :: Nil =>
+        val expectedStringSize = list.map(_.asInstanceOf[CONST_STRING].weight).sum
```
Comment (potan) [inline lang/shared/src/main/scala/com/wavesplatform/lang/v1/evaluator/ctx/impl/PureContext.scala:520]: Could we use weight of ARR object?

Comment (xrtm000) [reply lang/shared/src/main/scala/com/wavesplatform/lang/v1/evaluator/ctx/impl/PureContext.scala]: Yes

**Pull request merged**